File: android/app/src/main/java/com/HealthGauge/models/BLEDeviceModel.java
package com.HealthGauge.models;

import java.util.HashMap;

public class BLEDeviceModel {

    private String deviceAddress;
    private String deviceName;
    private String deviceRange;
    private String deviceUuids;
    private String deviceBondState;

    public String getDeviceBondState() {
        return deviceBondState;
    }

    public void setDeviceBondState(String deviceBondState) {
        this.deviceBondState = deviceBondState;
    }

    public String getDeviceUuids() {
        return deviceUuids;
    }

    public void setDeviceUuids(String deviceUuids) {
        this.deviceUuids = deviceUuids;
    }

    public String getDeviceAddress() {
        return deviceAddress;
    }

    public void setDeviceAddress(String deviceAddress) {
        this.deviceAddress = deviceAddress;
    }

    public String getDeviceName() {
        return deviceName;
    }

    public void setDeviceName(String deviceName) {
        this.deviceName = deviceName;
    }

    public String getDeviceRange() {
        return deviceRange;
    }

    public void setDeviceRange(String deviceRange) {
        this.deviceRange = deviceRange;
    }

    public void fromMap(HashMap map){
        deviceAddress = (String) map.get("address");
        deviceName = (String) map.get("name");
        deviceRange = (String) map.get("rssi");
        deviceUuids = (String) map.get("Uuids");
        deviceBondState = (String) map.get("BondState");

    }

    public HashMap toMap(){
        HashMap<String, Object> hashMap = new HashMap<>();
        hashMap.put("name",deviceName);
        hashMap.put("rssi",deviceRange);
        hashMap.put("address",deviceAddress);
        hashMap.put("Uuids",deviceUuids);
        hashMap.put("BondState",deviceBondState);

        return hashMap;
    }

}

File: lib/models/ble_device_model.dart
import 'package:health_gauge/utils/database_table_name_and_fields.dart';

class BLEDeviceModel {
  String? deviceName;
  String? deviceAddress;
  String? deviceRange;
  String? deviceUuids;
  String? deviceBondState;

  BLEDeviceModel(
      {this.deviceName,
        this.deviceAddress,
        this.deviceRange,
        this.deviceUuids,
        this.deviceBondState
      });

  BLEDeviceModel.fromMap(Map map) {
    if (check("name", map)) {
      deviceName = map["name"].toString();
    }
    if (check("rssi", map)) {
      deviceRange = map["rssi"].toString();
    }
    if (check("address", map)) {
      deviceAddress = map["address"].toString();
    }
    if (check("Uuids", map)) {
      deviceUuids = map["Uuids"].toString();
    }
    if (check("BondState", map)) {
      deviceBondState = map['BondState'].toString();
    }

  }

  BLEDeviceModel.clone(BLEDeviceModel device) {
    deviceName = device.deviceName;
    deviceAddress = device.deviceAddress;
    deviceRange = device.deviceRange;
    deviceUuids = device.deviceUuids ?? '';
    deviceBondState = device.deviceBondState ?? '0';

  }



  Map toMap() {
    return {
      "name": deviceName,
      "rssi": deviceRange,
      "address": deviceAddress,
      "Uuids": deviceUuids,
      "BondState": deviceBondState,

    };
  }

  check(String key, Map map) {
    if (map[key] != null) {
      if (map[key] is String && map[key] == "null") {
        return false;
      }
      return true;
    }
    return false;
  }
}
